addons/mail/models/mail_thread.py

```diff
@@ -2334,15 +2336,18 @@ def _process_attachments_for_post(self, attachments, attachment_ids, message_val
                 attachement_values_list.append(attachement_values)
 
                 # keep cid, name list and token synced with attachement_values_list length to match ids latter
-                attachement_extra_list.append((cid, name, token))
+                # Creating voice metadata if voice flag is set
+                attachement_extra_list.append((cid, name, token, voice))
 
             new_attachments = self.env['ir.attachment'].sudo().create(attachement_values_list)
             attach_cid_mapping, attach_name_mapping = {}, {}
-            for attachment, (cid, name, token) in zip(new_attachments, attachement_extra_list):
+            for attachment, (cid, name, token, voice) in zip(new_attachments, attachement_extra_list):
                 if cid:
                     attach_cid_mapping[cid] = (attachment.id, token)
                 if name:
                     attach_name_mapping[name] = (attachment.id, token)
+                if voice:
+                    attachment._post_add_create(voice=True)
```

Comment:
This is going to call "register as main attachment", although it is already managed. I would instead create discuss.voice.metadata manually (in batch) here and avoid calling "do a bit of everything" method as we are in an optimized processus here.